charactersheet/charactersheet/viewmodels/character/traits.js

```diff
@@ -0,0 +1,174 @@
+                }
+            }
+            Utility.array.updateElement(self.traits(), self.currentEditItem(), self.editItemIndex);
+        }
+
+        self.save();
+        self.currentEditItem(new Trait());
+        self.currentEditTracked(new Tracked());
+        self.modalOpen(false);
+        Notifications.trait.changed.dispatch();
+    };
+
+    self.selectPreviewTab = function() {
+        self.previewTabStatus('active');
+        self.editTabStatus('');
+    };
+
+    self.selectEditTab = function() {
+        self.editTabStatus('active');
+        self.previewTabStatus('');
+        self.editFirstModalElementHasFocus(true);
+    };
+
+    self.filteredAndSortedTraits = ko.computed(function() {
+        return SortService.sortAndFilter(self.traits(), self.sort(), null);
+    });
+
+    self.sortArrow = function(columnName) {
+        return SortService.sortArrow(columnName, self.sort());
+    };
+
+    self.sortBy = function(columnName) {
+        self.sort(SortService.sortForName(self.sort(),
+            columnName, self.sorts));
+    };
+
+    self.addTrait = function() {
+        var trait = self.blankTrait();
+        trait.characterId(CharacterManager.activeCharacter().key());
+        if (trait.isTracked()) {
+            trait.trackedId(uuid.v4());
+            self.addTracked(trait.trackedId(), trait.characterId(), self.blankTracked());
+        }
+        trait.save();
+        self.traits.push(trait);
+        self.blankTrait(new Trait());
+        self.blankTracked(new Tracked());
+    };
+
+    self.addTracked = function(uuid, characterId, tracked) {
+        var newTracked = new Tracked();
+        newTracked.characterId(characterId);
+        newTracked.trackedId(uuid);
+        newTracked.maxUses(tracked.maxUses());
+        newTracked.resetsOn(tracked.resetsOn());
+        newTracked.save();
+    };
+
+    self.clear = function() {
+        self.traits([]);
+    };
+
+    self.removeTrait = function(trait) {
+        if (trait.isTracked()) {
+            var tracked = PersistenceService.findFirstBy(
+                Tracked, 'trackedId', trait.trackedId());
+            tracked.delete();
+        }
+        self.traits.remove(trait);
+        trait.delete();
+        Notifications.trait.changed.dispatch();
+    };
+
+    self.editTrait = function(trait) {
+        self.editItemIndex = trait.__id;
+        self.currentEditItem(new Trait());
+        self.currentEditItem().importValues(trait.exportValues());
+        if (trait.isTracked()) {
+            self.currentEditTracked(PersistenceService.findFirstBy(Tracked, 'trackedId', trait.trackedId()));
+        }
+        self.modalOpen(true);
+    };
+
+    self.shortDescription = function(trait) {
+        return Utility.string.truncateStringAtLength(trait.description(), self.DESCRIPTION_MAX_LENGTH);
+    };
+
+    self.trackedPopoverText = function() {
+        return 'Tracked Traits are listed in Traiture Tracker.';
```

Comment:
I think this was an issue of using replace all? Let's just call it "Tracker" for now.